apps/web/views.py

```diff
@@ -35,29 +35,35 @@ def internal_server_error(request):
 
 
 @throttle_classes([AnonRateThrottle, ])
-@api_view(['POST', ])
+@api_view(['POST', 'GET'])
 @permission_classes((permissions.AllowAny,))
 def contact_us(request):
     try:
         user = User.objects.get(username=request.user)
         name = user.username
         email = user.email
-        request_data = {"name": name, "email": email}
-        request_data['message'] = request.data['message']
-        serializer = ContactSerializer(data=request_data)
+    except:
+        serializer = ContactSerializer(data=request.data)
         if serializer.is_valid():
             serializer.save()
             response_data = {'message': 'Your message has been successfully recorded. We will contact you shortly.'}
             return Response(response_data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
-    except:
-        serializer = ContactSerializer(data=request.data)
+
+    if request.method == 'POST':
+        request_data = {"name": name, "email": email}
+        request_data['message'] = request.data['message']
```

Comment:
On front end, there should be a check that message is mandatory, because if a user is trying to contact then he or she should definitely leave a message.

Also as far as throttling is concerned, I am not sure of what exactly should be the limit.

Anyways it will be nice if you use `.get` here.